Formula/idris2.rb

```diff
@@ -13,10 +12,12 @@ class Idris2 < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux: "fdd19a501c9b80da87fd4519ed43e9ed31647e17e03d582392d2b3d1517f7c74"
   end
 
-  depends_on "coreutils" => :build
   depends_on "gmp" => :build
   depends_on "chezscheme"
-  uses_from_macos "zsh" => :build, since: :mojave
+
+  on_high_sierra :or_older do
+    depends_on "zsh" => :build
+  end
```

Comment:
As mentioned in extended commit message, `zsh` shouldn't be needed on Linux as it is macOS-only alternative to `realpath` (`coreutils`) requirement. `uses_from_macos` will result in unnecessary Linux dependency, so use `on_high_sierra` DSL instead.

Ref: https://github.com/idris-lang/Idris2/commit/b457d15701f6f359d66c6e32af29561b44bfd81e